src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3116,32 +3152,67 @@ private Message GetSortedSetCombineAndStoreCommandMessage(SetOperation operation
             {
                 SetOperation.Intersect => RedisCommand.ZINTERSTORE,
                 SetOperation.Union => RedisCommand.ZUNIONSTORE,
+                SetOperation.Difference => RedisCommand.ZDIFFSTORE,
                 _ => throw new ArgumentOutOfRangeException(nameof(operation)),
             };
             if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFFSTORE cannot be used with weights or with aggregation.");
 
-            List<RedisValue>? values = null;
+            List<RedisValue>? values = new List<RedisValue>();
+            AddWeightsAndAggregation(ref values, weights, aggregate);
+            return new SortedSetCombineAndStoreCommandMessage(Database, flags, command, destination, keys, values?.ToArray() ?? RedisValue.EmptyArray);
+        }
+
+        private Message GetSortedSetCombineCommandMessage(SetOperation operation, RedisKey[] keys, double[]? weights, Aggregate aggregate, bool withScores, CommandFlags flags)
+        {
+            var command = operation switch
+            {
+                SetOperation.Intersect => RedisCommand.ZINTER,
+                SetOperation.Union => RedisCommand.ZUNION,
+                SetOperation.Difference => RedisCommand.ZDIFF,
+                _ => throw new ArgumentOutOfRangeException(nameof(operation)),
+            };
+            if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFF cannot be used with weights or with aggregation.");
```

Comment:
FWIW, this message is confusing to me because `Sum` _is_ one of the aggregations - not 100% sure on intent but happy to suggest better messaging once I read up on what this constraint is doing.

Reply:
Okay I get it, I propose instead of throwing here, we just say in docs that aggregation argument is ignored with the difference case and remove that portion of the check in methods - thoughts? The alternative I see is adding a `Aggregate.None`, which would have to be explicitly passed, seems simpler to just point out we ignore it.

Reply:
One more: if I understand correctly, should we check that weights has the same length as keys?

Reply:
In general, I support using redis exceptions. I mean that if redis already check the size of the weights and the keys, I don't think we should check it too.

Also, I removed the current exception, so now we get `StackExchange.Redis.RedisServerException` instead of `ArgumentException` which I think is ok :woman-shrugging::skin-tone-3:

Let me know what you think :)

Reply:
Now I'm thinking that if we do check the lengths, we can easily decide the length of the list so maybe it's better in this case :sweat_smile:

Reply:
If the error from the server makes it clear on how to fix, in total agreement - will take a peek when time allows today!

Reply:
Alrighty I poked at this locally, if Redis was more informative here I'd agree, but the error message users gets is "ERR Syntax error" with no details - so I opted to add some error checking here in latest to make it much easier to figure out what's been done wrong.